confd/pkg/backends/calico/client.go

```diff
@@ -1279,15 +1289,24 @@ func (c *client) getServiceExternalIPsKVPair(v3res *apiv3.BGPConfiguration, key
 
 func (c *client) getServiceLoadBalancerIPsKVPair(v3res *apiv3.BGPConfiguration, key interface{}, svcAdvertisement *bool) {
 	svcLoadBalancerIPKey := getBGPConfigKey("svc_loadbalancer_ips", key)
+	allowListKey := getBGPConfigKey("allowlist", key)
 
 	if v3res != nil && v3res.Spec.ServiceLoadBalancerIPs != nil && len(v3res.Spec.ServiceLoadBalancerIPs) != 0 {
 		ipCidrs := make([]string, len(v3res.Spec.ServiceLoadBalancerIPs))
 		for i, ipBlock := range v3res.Spec.ServiceLoadBalancerIPs {
 			ipCidrs[i] = ipBlock.CIDR
 		}
 		c.updateCache(api.UpdateTypeKVUpdated, getKVPair(svcLoadBalancerIPKey, strings.Join(ipCidrs, ",")))
+
+		if c.IsRouteReflector(c.rg.nodeName) {
+			c.updateCache(api.UpdateTypeKVUpdated, getKVPair(allowListKey, strings.Join(ipCidrs, ",")))
+		}
 	} else {
 		c.updateCache(api.UpdateTypeKVDeleted, getKVPair(svcLoadBalancerIPKey))
+
+		if c.IsRouteReflector(c.rg.nodeName) {
```

Comment:
This might be a non-issue in practice, but I think we should handle it regardless.

The scenario I'm thinking of:
- Start as a RR
- Update the cache with allowlist
- Disable the node as a RR
- Cache allowlist is no longer updated.

I think we might need similar checks to update the cache with the `allowlist` key when we switch from node -> RR and vice-versa.

Alternatively, we could always unconditionally update the allowlist key, and then in the template only render this section if we're a RR.

Reply:
What do you think about adding an else statement to the `isRouteReflector` check and deleting the `allowlist` cache entry if it's not.

Something like

```go
		if c.IsRouteReflector(c.rg.nodeName) {
			c.updateCache(api.UpdateTypeKVUpdated, getKVPair(allowListKey, strings.Join(ipCidrs, ",")))
		} else {
			c.updateCache(api.UpdateTypeKVDeleted, getKVPair(allowListKey))
		}

```

I'm pretty sure this will handle all the cases you mentioned

Reply:
I think the only problem with that is that the RouteReflector state is carried by the `Node` update, whereas the Service IPs are carried by BGPConfiguration. This means that any node can switch between being a RR and a normal node without this function even being executed (since it's only executed for BGPConfig updates)